hnn_core/network.py

```diff
@@ -1170,6 +1185,58 @@ def __repr__(self):
 
         return entr
 
+    def update_probability(self, probability):
+        """Remove/keep a random subset of connections.
+
+        Parameters
+        ----------
+        probability : float
+            Probability of connection between any src-target pair.
+            Defaults to 1.0 producing an all-to-all pattern.
+
+        Notes
+        -----
+        num_srcs and num_targets are not updated after pruning connections.
+        These variables are meant to describe the set of original connections
+        before they are randomly removed.
+
+        The probability attribute will store the most recent value passed to
+        this function. As such, this number does not accurately describe the
+        connections probability of the original set after successive calls.
+        """
+        _validate_type(probability, float, 'probability')
+        if probability <= 0.0 or probability >= 1.0:
+            raise ValueError('probability must be in the range [0,1]')
```

Comment:
```suggestion
            raise ValueError('probability must be in the range (0,1)')
```

since 0 and 1 are not included